File: Backend/src/services/news.py
from repositories.news import NewsRepository


class NewsService:
    def __init__(self, repository: NewsRepository | None = None):
        self.repository = repository or NewsRepository()

    async def get_news(
        self,
        ressort: str | None = None,
        regions: list[int] | None = None,
    ):
        return await self.repository.get_news(ressort=ressort, regions=regions)

File: Frontend/nimrag-frontend/src/services/news.ts
import { getJson } from './api'
import type { NewsQuery, NewsResponse } from '../types/news'

function buildNewsPath(query: NewsQuery = {}): string {
  const searchParams = new URLSearchParams()

  if (query.ressort) {
    searchParams.set('ressort', query.ressort)
  }
  if (query.regions?.length) {
    searchParams.set('regions', query.regions.join(','))
  }

  const queryString = searchParams.toString()
  return queryString.length > 0 ? `/news?${queryString}` : '/news'
}

export function getNews(query: NewsQuery = {}): Promise<NewsResponse> {
  return getJson<NewsResponse>(buildNewsPath(query))
}